src/core/OptionToken.sol

```diff
@@ -106,30 +109,11 @@ contract OptionToken is ERC1155, IOptionToken {
         if (msg.sender != address(grappa)) revert NoAccess();
     }
 
-    function _checkAccess(uint256 _tokenId) internal view {
-        (, uint40 productId, , , ) = TokenIdUtil.parseTokenId(_tokenId);
-        (, uint8 engineId, , , ) = ProductIdUtil.parseProductId(productId);
-        if (msg.sender != grappa.engines(engineId)) revert OT_Not_Authorized_Engine();
-    }
-
     /**
-     * @dev check if msg.sender is eligible for burning or minting certain token
+     * @dev check that the call is from authrized engine
      */
-    function _checkAccessAndTokenId(uint256 _tokenId) internal view {
-        (TokenType optionType, uint40 productId, uint64 expiry, uint64 longStrike, uint64 shortStrike) = TokenIdUtil
-            .parseTokenId(_tokenId);
-        (, uint8 engineId, , , ) = ProductIdUtil.parseProductId(productId);
+    function _checkEngineAccess(uint256 _tokenId) internal view {
```

Comment:
I think it makes sense to move this to grappa as well. 

Its intention is pretty straightforward, but what happens if the TokenId schema changes and TokenIdUtil is updated to support backwards compatibility. Grappa and Engines could also be upgraded but OptionToken will be out of sync.

Reply:
ok, and it will save some gas too because we can just have 1 external call ;)